src/etl/permits_extractor.py

```
@@ -0,0 +1,953 @@
+            else:
+                logger.info("No new records to process")
+                self._log_audit(
+                    status='success',
+                    records_processed=0,
+                    metadata={'last_cursor': max_cursor}
+                )
+            
+            return self.records_extracted
+            
+        except Exception as e:
+            logger.error(f"Extraction failed: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)